internal/ethapi/api.go

```diff
@@ -802,6 +802,10 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNr rpc.BlockNumb
 	if err := vmError(); err != nil {
 		return nil, 0, false, err
 	}
+	// If the timer caused an abort, return an appropriate error message
+	if evm.Cancelled() {
```

Comment:
Hmm, I guess there's a tiny data race here, whereby it *could* happen that a call finishes successfully, but then the timeout triggers before this line and a failure is returned... but that seems both improbable and not an issue (it's like winning the lottery, would have timed out for real most of the time).

Reply:
Yes, I know:

> This PR addresses thsi in a somewhat opportunistic way, since the timeout happens async in another goroutine. When we return from the actuall call, we check if the Cancel has been called on the evm. There is a tiny chance that the cancel was called after the execution finished correctly, but it's bettter to error on the safe side than how we do it now (and the chances of that happening are really quite slim).

Basically, the data-race is 'benevolent'. The worst that can happen is that it falsely says it was aborted, which is better than the other way around